shared/fileutil/fileutil.go

```diff
@@ -28,6 +27,51 @@ func ExpandPath(p string) (string, error) {
 	return filepath.Abs(path.Clean(os.ExpandEnv(p)))
 }
 
+// MkdirAll takes in a path, expands it if necessary, and looks through the
+// permissions of every directory along the path, ensuring we are not attempting
+// to overwrite any existing permissions. Finally, creates the directory accordingly
+// with standardized, Prysm project permissions. This is the static-analysis enforced
+// method for creating a directory programmatically in Prysm.
+func MkdirAll(pth string) error {
+	expanded, err := ExpandPath(pth)
+	if err != nil {
+		return err
+	}
+	exists, err := HasDir(pth)
+	if err != nil {
+		return err
+	}
+	if exists {
+		info, err := os.Stat(expanded)
+		if err != nil {
+			return err
+		}
+		if info.Mode().Perm() != params.BeaconIoConfig().ReadWriteExecutePermissions {
+			return errors.New("dir already exists with wrong permissions")
```

Comment:
Are you sure that `wrong` is the correct wording here? Maybe the permissions are correct and the new dir should be placed somewhere else.